### ISSUE TITLE
consider adding a symmetric version of associativity in the defintion of Category

### ISSUE THREAD

**Issue by HuStmpHrrr** [agda/agda-categories#59]
https://github.com/agda/agda-categories/blob/892b3f90f2ea796f48561b107fdfe8b5d1c88361/Categories/Category/Core.agda#L26

the idea of to define `assoc-sym`. the motivation is that this allows `op` of `op` to be definitionally equal to the original category. this is a handy treatment to avoid unnecessary setup.

**Comment by JacquesCarette** [comment]
You mean as an additional field?  If we can edide it 99% of the time, then that's worth considering.  Maybe make it implicit?

**Comment by HuStmpHrrr** [comment]
yeah an additional field. just flip the direction of `assoc`. this way, `op` would just need to swap `assoc` and `assoc-sym`, and that makes `op` of `op` definitionally equal to the original category. this helps the cases where opposite category is used.

**Comment by JacquesCarette** [comment]
It is definitely worth a try - if it works, that would be a nice thing to have. But I think I'll let you conduct that experiment...

**Comment by HuStmpHrrr** [comment]
OK so it looks good. for some reason `Category` needs to have `eta-equality` explicitly turned on. I thought this option is one by default.

**Comment by JacquesCarette** [comment]
Definitely something to document in the design decisions.
And yes, I thought `eta-equality` was on by default too - a bug?

**Comment by HuStmpHrrr** [comment]
not sure, I figure out a minimal reproducible case and ask if that's expected.

**Comment by HuStmpHrrr** [comment]
c.f. agda/agda#4142

**Comment by sstucki** [comment]
Sorry to show up late to the party, but could we add some comments in `Categories.Category.Core` to explain why `sym-assoc` is included? Newcomers to the library might be confused by the presence of this field.

**Comment by HuStmpHrrr** [comment]
sure that's not a problem. in fact, many libraries in Coq have this setup too.

**Comment by nachivpn** [comment]
Hello, I'm that confused newcomer! Why is it so important that the opposite category of the opposite category be definitionally equal to the original category?

**Comment by HuStmpHrrr** [comment]
> 
> 
> Hello, I'm that confused newcomer! Why is it important foe the opposite category of the opposite category to be definitionally equal to the original category?

in short, definitional equality is a blessing.

technically, it's more intuitive. think about a category `C` and its opposite. we usually think there are only two "sides" of a category. however, if `op` of `op` is just equivalent to `C`, then it means one can generate infinite number of categories by stacking `op`s. nonetheless, half of them is really *the same* as one and the other half is the same as another. so why would we prefer moving between `op` of `op` and `C` via a functor, the existence of which is purely caused by technical reasons? making things definitional is just cleaner. after all, you will have to show all limits, for example, is preserved by `op` of `op` etc, and that's just unnecessary burden to deal with.

**Comment by sstucki** [comment]
@HuStmpHrrr do you maybe have a concrete example from the library where it simplifies things?

I can come up with some example where it might, but I haven't actually encountered one in practice (in the library) yet.

**Comment by HuStmpHrrr** [comment]
here is one usage of that: https://github.com/agda/agda-categories/commit/b2cd7aa6c9c990de42ba881b619896cd0cd12dc2#diff-6218f7f546559e9086b91e1f79ed1fc5L63
